File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
    //     int count = 0;
    // int candidate = 0;

    // for (int num : nums) {
    //     if (count == 0) {
    //         candidate = num;
    //     }
    //     count += (num == candidate) ? 1 : -1;
    // }

    // return candidate;

    sort(nums.begin(), nums.end());

    int n = nums.size();
    return nums[n/2];
    }
};

File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let map1 = new Map();
    let n = nums.length;
    
    for(let num of nums){
        if(!map1.has(num)){
            map1.set(num,1)
        } else {
            map1.set(num,map1.get(num)+1)
        }
    }
    
    for(let [key,value] of map1){
        if(value > Math.floor(n/2)){
            return key;
        }
    }
};